packages/js/data/src/products/selectors.ts

```diff
@@ -122,38 +121,24 @@ export const isPending = (
 	return false;
 };
 
-export const getPermalinkParts = createRegistrySelector(
-	( select ) => ( state: ProductState, productId: number ) => {
-		const product = select( 'core' ).getEntityRecord(
-			'postType',
-			'product',
-			productId,
-			// @ts-expect-error query object is not part of the @wordpress/core-data types yet.
-			{
-				_fields: [
-					'id',
-					'permalink_template',
-					'slug',
-					'generated_slug',
-				],
-			}
-		);
-		if ( product && product.permalink_template ) {
-			const postName = product.slug || product.generated_slug;
+export const getPermalinkParts = ( state: ProductState, productId: number ) => {
+	const product = state.data[ productId ];
 
-			const [ prefix, suffix ] = product.permalink_template.split(
-				PERMALINK_PRODUCT_REGEX
-			);
+	if ( product && product.permalink_template ) {
+		const postName = product.slug || product.generated_slug;
 
-			return {
-				prefix,
-				postName,
-				suffix,
-			};
-		}
-		return null;
+		const [ prefix, suffix ] = product.permalink_template.split(
+			PERMALINK_PRODUCT_REGEX
+		);
+
+		return {
+			prefix,
+			postName,
+			suffix,
+		};
```

Comment:
@joshuatf I've been thinking about this... do you know why we split up the template and return the parts instead of just returning the generated permalink (by replacing the PERMALINK_PRODUCT_REGEX with the postName)? In our new UI, this seems like it would work well and would be a bit more elegant.

Reply:
I'm not entirely sure.  My best guess is that whoever added this was following the pattern in Gutenberg - https://github.com/WordPress/gutenberg/blob/64c69123ea57a3c342475c8a72d274d4cf9764a3/docs/reference-guides/data/data-core-editor.md?plain=1#L463

But this is also helpful for us dynamically updating the link in the UI prior to saving the post - https://github.com/woocommerce/woocommerce/blob/c977ed084bc903de6792bcce09d97f29e9b8511b/plugins/woocommerce-admin/client/products/fills/details-section/details-field-name.tsx#L81-L83

Reply:
I added this, and +1 on what @joshuatf, I followed the Gutenberg approach, and the regex is needed to generate the slug while a product hasn't been saved yet.

I did create an [issue](https://github.com/woocommerce/woocommerce/issues/34312) to add the `permalink_template` to the new API so we don't have to do a separate request. But I see you are already looking at that issue.